R/regularization.R

```diff
@@ -1,3 +1,59 @@
+#' Regularization via L1 norm
+#'
+#' Implements the L1-norm of the design's stage-two sample size function.
+#'
+#' @param s an \code{RegularizationScore}
+#' @param design a \code{TwoStageDesign}
+#'
+#' @rdname AverageN2-class
+#'
+#' @exportClass AverageN2
+setClass("AverageN2", representation(
+    dummy = 'logical'
+    ),
+    contains = "UnconditionalScore")
+
+
+
+#' @rdname AverageN2-class
+#' @export
+AverageN2 <- function() new("AverageN2", dummy = FALSE)
+
+#' Evaluation of a AverageN2 score
+#'
+#' @param specific logical, flag for switching to design-specific implementation.
+#' @param ... further optimal arguments
+#'
+#' @describeIn AverageN2 generic implementation of evaluating a smoothness
+#'     score. Uses adaptive Gaussian quadrature for integration and might be
+#'     more efficiently implemented by specific \code{TwoStageDesign}-classes
+#'     (cf. \code{\link{.evaluate}}).
+setMethod("evaluate", signature("AverageN2", "TwoStageDesign"),
+          function(s, design, specific = TRUE, ...) {
+              if (specific) { # use design-specific implementation
+                  return(.evaluate(s, design, ...))
+              } else {
+                  res <- stats::integrate(
+                      function(x) n2(design, x),
+                      design@c1f,
+                      design@c1e
+                  )$value
+                  res <- res / (design@c1e - design@c1f)
+                  return(res)
+              }
+          }
+)
+
+
+# not user facing!
+setMethod(".evaluate", signature("AverageN2", "TwoStageDesign"),
+          function(s, design, ...) {
+              sum(design@n2_pivots * design@weights) / (design@c1e - design@c1f)
```

Comment:
still not happy with that (cf. above)

Reply:
I changed it, but if we get runtime problems, we can thinking of returning to this one